src/Sulu/Component/CustomUrl/Manager/CustomUrlManager.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\CustomUrl\Manager;
+
+use Ferrandini\Urlizer;
+use Sulu\Component\CustomUrl\Document\CustomUrlDocument;
+use Sulu\Component\CustomUrl\Repository\CustomUrlRepository;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\PathBuilder;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * Manages custom-url documents and their routes.
+ */
+class CustomUrlManager implements CustomUrlManagerInterface
+{
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var CustomUrlRepository
+     */
+    private $customUrlRepository;
+
+    /**
+     * @var PathBuilder
+     */
+    private $pathBuilder;
+
+    public function __construct(
+        DocumentManagerInterface $documentManager,
+        CustomUrlRepository $customUrlRepository,
+        PathBuilder $pathBuilder
+    ) {
+        $this->documentManager = $documentManager;
+        $this->customUrlRepository = $customUrlRepository;
+        $this->pathBuilder = $pathBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($webspaceKey, array $data, $locale = null)
+    {
+        $document = $this->documentManager->create('custom_urls');
+        $this->bind($document, $data, $locale);
+
+        $this->documentManager->persist(
+            $document,
+            $locale,
+            [
+                'parent_path' => $this->getItemsPath($webspaceKey),
+                'node_name' => Urlizer::urlize($document->getTitle()),
```

Comment:
Hmm. Snippets use the AutoFiling feature or similar no?


On 18 January 2016 14:15:51 GMT+00:00, Wachter Johannes <notifications@github.com> wrote:
>> +    }
>> +
>> +    /**
>> +     * {@inheritdoc}
>> +     */
>> +    public function create($webspaceKey, array $data, $locale =
>null)
>> +    {
>> +        $document = $this->documentManager->create('custom_urls');
>> +        $this->bind($document, $data, $locale);
>> +
>> +        $this->documentManager->persist(
>> +            $document,
>> +            $locale,
>> +            [
>> +                'parent_path' => $this->getItemsPath($webspaceKey),
>> +                'node_name' =>
>Urlizer::urlize($document->getTitle()),
>
>i cannot find this functionality and i cannot find a configuration for
>the path!
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/sulu-io/sulu/pull/1927/files#r50002453

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.